packages/app/src/embed/components/Files/index.js

```diff
@@ -83,4 +79,80 @@ function Files({
   );
 }
 
+function Directory({
+  directory,
+  currentModuleId,
+  setCurrentModule,
+  template,
+  entry,
+  modules,
+  directories,
+  depth,
+}) {
+  /** directory should be open by default if currentModule is inside it */
+
+  const currentModuleTree = getCurrentModuleTree(
+    modules,
+    directories,
+    currentModuleId
+  );
+
+  let openByDefault = false;
+  if (currentModuleTree.find(module => module.id === directory.id)) {
+    openByDefault = true;
+  }
+
+  const [open, setOpen] = React.useState(openByDefault);
+
+  return (
+    <div>
+      <File
+        id={directory.id}
+        shortid={directory.shortid}
+        title={directory.title}
+        type={open ? 'directory-open' : 'directory'}
+        depth={depth}
+        setCurrentModule={setCurrentModule}
+        onClick={_ => setOpen(!open)}
+      />
+      {open ? (
+        <Files
+          modules={modules}
+          directories={directories}
+          directoryId={directory.shortid}
+          depth={depth + 1}
+          setCurrentModule={setCurrentModule}
+          currentModule={currentModuleId}
+          template={template}
+          entry={entry}
+        />
+      ) : null}
+    </div>
+  );
+}
+
 export default Files;
+
+/** Utils to help identify module tree */
+
+const getParentDirectory = (directories, child) => {
+  return directories.find(directory => {
```

Comment:
This could be an implicit return

Reply:
Ah, I'm really conflicted about implicit returns for really long statements

The latter is shorter, but I'm not sure if it's easier to _read_/understand 🤷‍♀ 

![Screenshot 2019-09-04 at 11 13 53 AM](https://user-images.githubusercontent.com/1863771/64242056-27bd1500-cf05-11e9-882c-ba424fb2dd6b.png)

@SaraVieira @Saeris how do you folks feel about this?

Reply:
Personally I always go with implicit returns. I'm fine either way.

Reply:
> Personally I always go with implicit returns. I'm fine either way.

I try to break the rule when needed 😅, but I'm fine either way too 👍. Changed it to implicit returns